src/libraries/Microsoft.PowerFx.Connectors/Public/CdpRecordType.cs

```diff
@@ -77,12 +82,23 @@ public override bool TryGetFieldType(string fieldName, out FormulaType type)
             }
         }
 
-#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
         public override bool Equals(object other)
```

Comment:
>  public override bool Equals(object other) [](http://example.com/codeflow?start=6&length=43)

Can we avoid implementing this and continue to throw?
Semantics on record comparison are not clear.  
If caller just needs a null check, they can use 'is null' or 'object.ReferenceEquals'. 

If they're doing this for any other reason - it's probably a bug in their code.

Reply:
No, Ajeet is comparing with null and is observing an exception
I agree ReferenceEquals is a valid workaround but this isn't what I'd call a clean approach so I prefer implementing it

Reply:
If caller just needs a null check, they can use 'is null' or 'object.ReferenceEquals'. 

Our caller doesn't need this and PowerFx does not define equality semantics on records (so anybody calling this probably means a bug in their code) - so this is a liability with minimal benefit. 

I see a lot of conditions and a deep call stack when I f12 into this,. 
If you do proceed with this implementation, please make sure we have 100% code coverage on the code path.